components/sup/src/manager/service/mod.rs

```diff
@@ -618,8 +622,9 @@ impl Service {
             self.initialize();
             if self.initialized {
                 self.start(launcher);
-                self.post_run();
             }
+        } else if self.health_check == HealthCheck::Ok && !self.post_run {
```

Comment:
I think pulling this conditional out into a function named to capture the intention ("keep running a post-run hook if we're healthy and haven't run the post-run hook yet") will help to clarify the desired behavior, and make this code read more easily.